include/openthread/border_routing.h

```diff
@@ -498,6 +498,35 @@ void otBorderRoutingDhcp6PdSetEnabled(otInstance *aInstance, bool aEnabled);
  */
 otBorderRoutingDhcp6PdState otBorderRoutingDhcp6PdGetState(otInstance *aInstance);
 
+/**
+ * Function pointer callback which is invoked when state of request/release a prefix via DHCPv6 Prefix Delegation (PD)
+ * for Thread interface changed.
+ *
+ * When the DHCPv6 PD state changed, we need to notify another process running on the OS whether they should request
+ * an prefix via DHCPv6 PD for Thread interface. When aRequest is false, the process must release the prefix lease.
+ *
+ * @param[in] aRequest  Whether to request RA.
+ * @param[in] aContext  A pointer to arbitrary context information.
+ *
+ */
+typedef void (*otBorderRoutingRequestDhcp6PdCallback)(bool aRequest, void *aContext);
+
+/**
+ * Set the callback when state of request/release a prefix via DHCPv6 Prefix Delegation (PD)
+ * for Thread interface changed.
+ *
+ * Subsequent calls to this function will replace the previously set callback
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ * @param[in] aCallback  A pointer to a function that is called when request/release PD state changes.
```

Comment:
@param[in]  aCallback   A pointer to a function that is called whenever  the state of a prefix request or release  changes.
```suggestion
 * @param[in] aCallback  A pointer to a function that is called when request/release PD state changes.
```